src/cli/load.js

```diff
@@ -5,12 +5,29 @@ import type {Command} from "./command";
 import {loadInstanceConfig, pluginDirectoryContext} from "./common";
 import {LoggingTaskReporter} from "../util/taskReporter";
 import * as pluginId from "../api/pluginId";
+import {Plugin, PluginDirectoryContext} from "../api/plugin";
+import {isDirEmpty} from "../util/disk";
+import fs from "fs-extra";
 
 function die(std, message) {
   std.err("fatal: " + message);
   return 1;
 }
 
+type PluginLoadContext = {|
+  +plugin: Plugin,
+  +dirContext: PluginDirectoryContext,
+  +taskReporter: LoggingTaskReporter,
+  +task: string,
+|};
+
+function loadPlugin(ctx: PluginLoadContext) {
+  const {plugin, dirContext, taskReporter, task} = ctx;
+  return plugin
+    .load(dirContext, taskReporter)
+    .then(() => taskReporter.finish(task));
+}
```

Comment:
This can be more concisely (and imho more clearly) expressed as a
closure, capturing the environment rather than reifying it:

```javascript
    const loadPlugin = () =>
      plugin
        .load(dirContext, taskReporter)
        .then(() => taskReporter.finish(task));
    const promise = loadPlugin()
      .catch((e) => {
        if (!cacheEmpty.get(name)) {
          // clear the cache and try again
          fs.emptyDirSync(dirContext.cacheDirectory());
          return loadPlugin();
        }
        throw e;
      })
      .catch((e) => {
        console.error(e);
        failedPlugins.push(name);
      });
```

Net diff is −16 lines locally, plus you drop an import.